### ISSUE TITLE
Re-License from MIT to ?

### ISSUE THREAD

**Issue by notpeter** [tldr-pages/tldr#1076]
This can of worms was opened in #1048 perhaps we should relicense with a documentation specific license rather than the current MIT license. This issue applies only to tldr pages, not the tldr clients.

Some potential options: 
- [CC0 (Public Domain)](http://choosealicense.com/licenses/cc0-1.0/)
- [CC-BY-4.0](http://choosealicense.com/licenses/cc-by-4.0/)
- [CC-BY-SA-4.0 (Share Alike)](http://choosealicense.com/licenses/cc-by-sa-4.0/)
- Stick with [MIT](http://choosealicense.com/licenses/mit/)

Any re-licensure could be accomplished reaching out to all previous committers to get them to take 30 seconds to agree to the [Contributor License Agreement wizard proposed in #1048](https://github.com/tldr-pages/tldr/issues/1048).  If original contributors can't be reached/don't agree we would just have to re-write/replace the examples they wrote.

**Comment by ostera** [comment]
It'd be important to move away from MIT so people can't sue the individual maintainers/contributors if they happen to fuck up their computers/lives because of a miscopied command.

But then I'm not well read on CC* licenses either.

**Comment by waldyrious** [comment]
I'm not sure I understand your issue with MIT, @ostera, given the disclaimer it contains (which is pretty standard in most licenses):

> THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
SOFTWARE.

In any case, I do agree we should move away from MIT for the pages content, which is more suited to a CC-type license.

I personally would prefer a license with similar properties than MIT (attribution), and furthermore with copyleft-like properties (share-alike). Hence my suggestion would be [CC-BY-SA](https://creativecommons.org/licenses/by-sa/4.0/) ([4.0](https://creativecommons.org/share-your-work/licensing-considerations/version4/), [International](https://en.wikipedia.org/wiki/Creative_Commons_license#Version_4.0_and_international_use)).

**Comment by waldyrious** [comment]
To track the progress in having the contributor license agreement (CLA) signed by the major contributors (which isn't strictly required for relicensing, but would make it considerably easier), I'll list below the top 100 [contributors to the repository](https://github.com/tldr-pages/tldr/graphs/contributors), and whether they've [signed the CLA](https://cla-assistant.io/tldr-pages/tldr). The top 10 who haven't yet signed are @-pinged.

- [x] waldyrious
- [ ] @igorshubovych
- [ ] @oxguy3
- [ ] @rubenvereecken
- [ ] @rprieto
- [ ] @notpeter
- [x] agnivade
- [ ] @onesuper
- [x] ostera
- [x] alx741
- [ ] @85pando
- [ ] @Cvetomird91
- [ ] @felixonmars
- [x] sbrl
- [x] ericbn
- [x] Dahie
- [ ] @dylanrees
- [x] joelhy
- [ ] steelywing
- [ ] lord63
- [ ] Like-all
- [ ] naxoc
- [ ] akoenig
- [ ] sethwoodworth
- [x] slash3b
- [ ] britter
- [ ] chirag64
- [ ] mumumu
- [x] chuanconggao
- [x] pearcem0
- [ ] bobstrecansky
- [x] zdroid
- [ ] Larry850806
- [x] lagelalegal
- [x] AloisMahdal
- [x] liuderchi
- [ ] d33tah
- [ ] egilkh
- [x] eliotsykes
- [ ] bripkens
- [ ] matthewgao
- [x] Fallon9111
- [ ] Duologic
- [ ] martypenner
- [ ] fordhurley
- [ ] Leandros
- [x] denis-sokolov
- [x] pindexis
- [ ] kuanyui
- [ ] jegtnes
- [ ] pranavraja
- [ ] fuerbringer
- [ ] hugovk
- [ ] sn0w
- [ ] ntodd
- [ ] jezeniel
- [ ] jlinder
- [ ] appleboy
- [ ] AdamSulucz
- [ ] jkbrzt
- [ ] shadowfax92
- [ ] ignazioc
- [ ] therealmarv
- [ ] rakeshdas1
- [ ] shybovycha
- [ ] rselwyn
- [ ] dbrgn
- [ ] dyng
- [ ] jlems
- [ ] sfarzy
- [x] eddieantonio
- [ ] kmoe
- [ ] rcorrie
- [ ] byjord
- [ ] phillips1012
- [ ] RoryCrispin
- [ ] axelcdv
- [x] fluxw42
- [x] jonboiser
- [ ] tjwudi
- [ ] kumon
- [ ] jeanleonino
- [ ] rnas
- [ ] gwanii
- [ ] charlestang
- [ ] lmount
- [ ] vpilot
- [ ] SShrike
- [ ] AgamAgarwal
- [ ] zhengqm
- [x] yasharmaster
- [ ] aicioara
- [ ] maxbeier
- [ ] livioso
- [ ] emileber
- [ ] smevawala
- [ ] robyoung
- [ ] atweiden
- [ ] matkoniecz
- [ ] cnu

I'll periodically update the list, and annotate when the list was last updated. Last update: 2016-10-16.

Note: ideally we would cross-reference this list with the blame information of the pages to get more granular info (i.e. we could relicense the pages whose contributors have all signed the CLA), but for now this is a start.

**Comment by 85pando** [comment]
I have not really read the MIT license in detail, but is it not a take-it-and-do-as-you-wish-license? Staying in this spirit it would be more appropriate to use CC0.

Still i myself like the CC-BY license best. It forces the license to stick with any derivates and will therefore also pertain credit to the authors. I favor it over the CC-BY-SA license because this practically forbids use with any other license thus making it potentially impossible to use (e.g. not compatible with CC-BY-SA-NC).

**Comment by waldyrious** [comment]
@85pando yes, in a way the MIT is part of the permissive family of licences, considering that the only condition it imposes is preservation of the license and copyright notice; CC-BY is a close approximation in terms of content, indeed, while CC-BY-SA is closer to the copyleft (protective) licenses, albeit still fairly permissive.

I'm not sure I understand what you mean by "potentially impossible to use (e.g. not compatible with CC-BY-SA-NC)", though. Can you clarify, perhaps with an example?

**Comment by 85pando** [comment]
> I'm not sure I understand what you mean by "potentially impossible to use (e.g. not compatible with CC-BY-SA-NC)", though. Can you clarify, perhaps with an example?

The most simple example would be:

1. Person A licenses an Image under CC-BY-SA.
2. Person B licenses an Image under CC-BY-SA-NC.
3. Person C wants to Remix both of them into one Image. This is not possible, as both licenses forbid use of the other license (i.e. Image A cannot be put under "NC", because it would mean less rights for the derivative work, Image B cannot be put under "not NC" because it would mean more rights for the derivative work.

**Comment by waldyrious** [comment]
Should we worry about that case though, considering that -NC creative commons licenses do not conform to the [definition of "open license"](http://opendefinition.org/licenses/) as described in our CLA? I'm not saying that we should actively attempt to block the applicability of such licenses, but I don't think we should go out of the way to accommodate them either.

**Comment by 85pando** [comment]
This also applies to other open licenses. CC-SA licenses and until judged by a court noone can really say if they are compatible (I'm a layman here myself, but thats the opinion I encounter most often).

**Comment by waldyrious** [comment]
I'm pretty sure the legal team at Creative Commons has taken care to ensure that these licenses are shareable and comply with their original vision, which is precisely to avoid the silos that exist with traditional copyright model.

Furthermore, the knowledgeable folks behind the curation of which licenses qualify as open must surely have taken these concerns into account. The fact that they've excluded -NC licenses from the list gives some indication that they did consider the actual merits of the CC licenses rather than just rubber-stamp them. For these reasons, I certainly feel comfortable deferring to their assessment on this issue.

**Comment by rprieto** [comment]
Hi, to answer one of the questions in #1048: I don't think I need to be mentioned in the LICENSE at all. This was just the way it was setup at first with my 0 knowledge of licenses :) I'd rather see `Copyright: The TLDR team at https://github.com/tldr-pages`.

**Comment by oxguy3** [comment]
Hey, just wanted to let y'all know that I went ahead and signed the CLA. I'm fine with any license, but a more permissive license is definitely preferable. The MIT License would be my top choice (but CC0/public domain are fine as well).

**Comment by waldyrious** [comment]
Thanks for the clarification @rprieto. I'll make the change in the license file then. @oxguy3 thanks for signing also :) we're slowly getting closer to the point where a license change will become practical.

**Comment by waldyrious** [comment]
@85pando I guess it had never dawned on me that CC-BY vs. CC-BY-SA is actually quite comparable to MIT vs. GPL. [This comment](https://github.com/github/choosealicense.com/pull/470#issuecomment-270193352) made me see things differently, and yeah, I guess it would be more in line with the spirit of the project's current licensing to move the pages to CC-BY rather than CC-BY-SA. (CC0 would be closer to WTFPL or Unlicense on the software side, than to MIT).

Given the above I swap my preference to CC-BY as a potential relicensing target.

**Comment by d33tah** [comment]
@waldyrious so, am I supposed to make a specific choice or something? I flicked through this discussion and it looks like you guys hadn't decided what you want yet. Let me know when you do and I'll tell you if I like it - quite possibly I'll say that I do.

**Comment by waldyrious** [comment]
@d33tah you were pinged (as were the others) about signing the CLA, precisely to prevent the license transition from depending on synchronous agreement of all the hundreds of involved contributors, which would be a nightmare to manage (case in point, look how long it's taken us to gather the existing signatures in the asynchronous CLA process).

The exact licence we'll change into hasn't been set in stone yet, but the discussion seems to be coalescing between CC-BY or CC0. In any case, any past contributions will remain perpetually available under the original license (MIT), so nothing is lost regardless of the license we eventually decide on (in case that was a concern regarding signing of the CLA).

**Comment by waldyrious** [comment]
@onesuper, @Cvetomird91, @dylanrees, @naxoc, @britter, @mumumu, @Larry850806, @egilkh, @bripkens and @matthewgao: we'd really appreciate if you could take a look at the [tldr-pages CLA](https://cla-assistant.io/tldr-pages/tldr) (adopted in #1048), sign it if you agree with it, or let us know if you have any objections. Thanks!

**Comment by britter** [comment]
@waldyrious done!

**Comment by naxoc** [comment]
Done!

**Comment by waldyrious** [comment]
Thanks guys, for the fast response!

@fordhurley, @Leandros, @kuanyui, @jegtnes, @pranavraja, @hugovk, @sn0w, @ntodd, @jezeniel, @jlinder, @appleboy, @AdamSulucz, @jkbrzt, @shadowfax92, @ignazioc, @therealmarv, @rakeshdas1, @shybovycha, @rselwyn, @dbrgn, @dyng, @jlems, @sfarzy, @kmoe, @rcorrie, @byjord, @phillips1012, @RoryCrispin, @axelcdv, @tjwudi, @kumon, @jeanleonino, @rnas, @gwanii, @charlestang, @lmount, @vpilot, @SShrike, @AgamAgarwal, @zhengqm, @aicioara, @livioso, @emileber:

As mentioned above, we'd really appreciate if you could take a look at the [tldr-pages CLA](https://cla-assistant.io/tldr-pages/tldr) (adopted in #1048), sign it if you agree with it, or let us know if you have any objections. If you don't have any comments, please just thumbs-up this comment instead of writing below, so we can avoid making this thread too long. Thanks in advance :)

**Comment by kuanyui** [comment]
I'm here to declare: I'm to abandon all rights of my contribution for this repository; so please feel free to change license to whatever you want in the future and don't notice me again. :-)

**Comment by tjwudi** [comment]
Signed. I delegate these sort of decision to you guys since my contribution were pretty small ;)

**Comment by rcorrie** [comment]
signed

**Comment by Duologic** [comment]
signed, for what it's worth :)

**Comment by waldyrious** [comment]
@Duologic are you sure? I can't find your name on cla-assistant's list of signataries. Could you go to https://cla-assistant.io/tldr-pages/tldr to confirm, please?

**Comment by Duologic** [comment]
Ah, sorry, I thought it would only be needed to agree in this issue. I was not planning on scrolling back and reading everything, tl;dr;.... :D

**Comment by waldyrious** [comment]
Heheh no problem :) it's understandable since the thread is quite long already...

**Comment by waldyrious** [comment]
Re-pinging @Cvetomird91, @mumumu, @Larry850806, @egilkh, @matthewgao, @fordhurley, @pranavraja, @jezeniel, @jlinder, @AdamSulucz, @jkbrzt, @shybovycha, @jlems, @byjord, @RoryCrispin, @axelcdv, @jeanleonino, @rnas, @gwanii, @zhengqm, and @livioso; pinging (for the first time) @smevawala, @robyoung, @atweiden, @matkoniecz, and @cnu:

We need your collaboration :) please visit https://cla-assistant.io/tldr-pages/tldr to read the tldr-pages contributor license agreement and confirm your signature of it, or comment below (or react with a :-1: to this comment) if you have any objections. Thanks!

**Comment by egilkh** [comment]
Done.

Sorry for the long wait. Kinda turned off most notifications lately.

**Comment by waldyrious** [comment]
It's quite alright, @egilkh -- we all have periods where we need to step aside for a while :) Thanks for signing!

**Comment by zdroid** [comment]
Is this even worth the hassle? Seems like stale issue, but I'd go with closing it.

**Comment by waldyrious** [comment]
@zdroid I think it was quite worth the hassle, actually, as we managed to collect a pretty wide set of feedback from the community, from which we can judge consensus on this matter — and if I'm reading the thread correctly, there's a clear consensus that this change is OK with the vast majority of contributors.

We also managed to have many of the contributors from before we adopted the CLA to sign it, which was also a positive outcome.

That said, I do agree that we have collected sufficient feedback to make a decision. Based on the discussion above, and considering that the current license (MIT) does allow relicensing already, I will officially propose relicensing the pages to the closest Creative-Commons license to MIT's properties, which would be CC-BY. I'll open a PR which, once merged, will finally close this issue! 🎉
